workload/rand: TestRandRun failed

workload/rand.TestRandRun [failed](https://tanzanite.cluster.engflow.com/invocations/default/35237fc8-2de3-472d-adeb-1a5f6dc06c50?testReportRun=2&testReportShard=1&testReportAttempt=1#targets-Ly9wa2cvd29ya2xvYWQvcmFuZDpyYW5kX3Rlc3Q=) on release-24.1 @ [3f9e0dbd11601ab97f8a3878a0fac07ca40ba2d8](https://github.com/cockroachdb/cockroach/commits/3f9e0dbd11601ab97f8a3878a0fac07ca40ba2d8):

Fatal error:

```
panic: test timed out after 19m57s
running tests:
	TestRandRun (19m57s)

```
Stack: 

```
goroutine 33011 [running]:
testing.(*M).startAlarm.func1()
	GOROOT/src/testing/testing.go:2366 +0x265
created by time.goFunc
	GOROOT/src/time/sleep.go:177 +0x45
```
<details><summary>Log preceding fatal error</summary>
<p>

```
* github.com/cockroachdb/cockroach/pkg/util/admission.(*tokenAllocationTicker).tick(...)
* 	github.com/cockroachdb/cockroach/pkg/util/admission/io_load_listener.go:425
* github.com/cockroachdb/cockroach/pkg/util/admission.(*StoreGrantCoordinators).SetPebbleMetricsProvider.func1()
* 	github.com/cockroachdb/cockroach/pkg/util/admission/grant_coordinator.go:115 +0xda
* created by github.com/cockroachdb/cockroach/pkg/util/admission.(*StoreGrantCoordinators).SetPebbleMetricsProvider in goroutine 15
* 	github.com/cockroachdb/cockroach/pkg/util/admission/grant_coordinator.go:109 +0x505
* 
* goroutine 557 [chan send, 1 minutes]:
* github.com/cockroachdb/cockroach/pkg/sql/sqlstats/insights.newConcurrentBufferIngester.func2(0xc003848460?)
* 	github.com/cockroachdb/cockroach/pkg/sql/sqlstats/insights/pkg/sql/sqlstats/insights/ingester.go:208 +0x118
* github.com/cockroachdb/cockroach/pkg/sql/contention/contentionutils.(*ConcurrentBufferGuard).syncLocked(0xc003848460)
* 	github.com/cockroachdb/cockroach/pkg/sql/contention/contentionutils/concurrent_buffer_guard.go:158 +0x52
* github.com/cockroachdb/cockroach/pkg/sql/contention/contentionutils.(*ConcurrentBufferGuard).ForceSync(0xc003848460)
* 	github.com/cockroachdb/cockroach/pkg/sql/contention/contentionutils/concurrent_buffer_guard.go:123 +0x87
* github.com/cockroachdb/cockroach/pkg/sql/sqlstats/insights.(*concurrentBufferIngester).Start.func2({0xc0021c9008?, 0xc0016b3380?})
* 	github.com/cockroachdb/cockroach/pkg/sql/sqlstats/insights/pkg/sql/sqlstats/insights/ingester.go:120 +0x65
* github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).RunAsyncTaskEx.func2()
* 	github.com/cockroachdb/cockroach/pkg/util/stop/stopper.go:485 +0x263
* created by github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).RunAsyncTaskEx in goroutine 15
* 	github.com/cockroachdb/cockroach/pkg/util/stop/stopper.go:476 +0x69d
* 
* goroutine 1840 [select, 5 minutes]:
* database/sql.(*DB).connectionOpener(0xc0054f4340, {0xb0c0420, 0xc004dbdb30})
* 	GOROOT/src/database/sql/sql.go:1246 +0xed
* created by database/sql.OpenDB in goroutine 15
* 	GOROOT/src/database/sql/sql.go:824 +0x2d4
* 
* goroutine 33004 [sync.Cond.Wait, 1 minutes]:
* sync.runtime_notifyListWait(0xc007fde0a8, 0x0)
* 	GOROOT/src/runtime/sema.go:569 +0x159
* sync.(*Cond).Wait(0xc007fde098)
* 	GOROOT/src/sync/cond.go:70 +0x75
* github.com/cockroachdb/pebble/record.(*flusherCond).Wait(...)
* 	github.com/cockroachdb/pebble/record/external/com_github_cockroachdb_pebble/record/log_writer.go:369
* github.com/cockroachdb/pebble/record.(*LogWriter).flushLoop(0xc007fde000, {0xc005d69f40?, 0x7acebe0?})
* 	github.com/cockroachdb/pebble/record/external/com_github_cockroachdb_pebble/record/log_writer.go:643 +0xb79
* runtime/pprof.Do({0xb0c0308, 0xfb5b440}, {{0xc000cbf480?, 0x75adc70?, 0x7ff07ec3f900?}}, 0xc00382ffc0)
* 	GOROOT/src/runtime/pprof/runtime.go:51 +0x118
* github.com/cockroachdb/pebble/record.NewLogWriter.func2()
* 	github.com/cockroachdb/pebble/record/external/com_github_cockroachdb_pebble/record/log_writer.go:562 +0x92
* created by github.com/cockroachdb/pebble/record.NewLogWriter in goroutine 32834
* 	github.com/cockroachdb/pebble/record/external/com_github_cockroachdb_pebble/record/log_writer.go:561 +0x8e5
* 
* goroutine 33003 [select]:
* github.com/cockroachdb/pebble/vfs.(*diskHealthCheckingFile).startTicker.func1()
* 	github.com/cockroachdb/pebble/vfs/external/com_github_cockroachdb_pebble/vfs/disk_health.go:243 +0x19b
* created by github.com/cockroachdb/pebble/vfs.(*diskHealthCheckingFile).startTicker in goroutine 32834
* 	github.com/cockroachdb/pebble/vfs/external/com_github_cockroachdb_pebble/vfs/disk_health.go:238 +0xad
* 
*
```
</p>
</details>

Parameters:
 - <code>attempt=1</code>
 - <code>race=true</code>
 - <code>run=2</code>
 - <code>shard=1</code>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
/cc @cockroachdb/test-eng @cockroachdb/sql-foundations
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestRandRun.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)
</sub>


Jira issue: CRDB-37614

--- a/pkg/workload/rand/BUILD.bazel
+++ b/pkg/workload/rand/BUILD.bazel
@@ -35,6 +35,7 @@ go_test(
         "//pkg/sql/sem/tree",
         "//pkg/sql/types",
         "//pkg/testutils/serverutils",
+        "//pkg/testutils/skip",
         "//pkg/testutils/sqlutils",
         "//pkg/util/leaktest",
         "//pkg/util/log",

--- a/pkg/workload/rand/rand_test.go
+++ b/pkg/workload/rand/rand_test.go
@@ -24,6 +24,7 @@ import (
 	"github.com/cockroachdb/cockroach/pkg/sql/sem/tree"
 	"github.com/cockroachdb/cockroach/pkg/sql/types"
 	"github.com/cockroachdb/cockroach/pkg/testutils/serverutils"
+	"github.com/cockroachdb/cockroach/pkg/testutils/skip"
 	"github.com/cockroachdb/cockroach/pkg/testutils/sqlutils"
 	"github.com/cockroachdb/cockroach/pkg/util/leaktest"
 	"github.com/cockroachdb/cockroach/pkg/util/log"
@@ -46,6 +47,8 @@ func TestMain(m *testing.M) {
 func TestRandRun(t *testing.T) {
 	defer leaktest.AfterTest(t)()
 	defer log.Scope(t).Close(t)
+
+	skip.UnderDuress(t, "random test, can time out under duress")
 
 	ctx, cancel := context.WithCancel(context.Background())
 	defer cancel()

